SW_Output_get_functions.c

```diff
@@ -188,135 +188,220 @@ void get_none(OutPeriod pd)
 
 	RealD
-		*p = p_OUT[eSW_CO2Effects][pd],
-		*psd = p_OUTsd[eSW_CO2Effects][pd];
+		*p = p_OUT[eSW_Biomass][pd],
+		*psd = p_OUTsd[eSW_Biomass][pd];
 
+	// scale total biomass by fCover to obtain 100% total cover biomass
 	ForEachVegType(k)
 	{
-		biomass_total += vo->veg[k].biomass;
-		biolive_total += vo->veg[k].biolive;
-	}
-
-	do_running_agg(p, psd, iOUT(0, pd), Globals.currIter, vo->veg[SW_GRASS].biomass);
-	do_running_agg(p, psd, iOUT(1, pd), Globals.currIter, vo->veg[SW_SHRUB].biomass);
-	do_running_agg(p, psd, iOUT(2, pd), Globals.currIter, vo->veg[SW_TREES].biomass);
-	do_running_agg(p, psd, iOUT(3, pd), Globals.currIter, vo->veg[SW_FORBS].biomass);
-	do_running_agg(p, psd, iOUT(4, pd), Globals.currIter, biomass_total);
-	do_running_agg(p, psd, iOUT(5, pd), Globals.currIter, vo->veg[SW_GRASS].biolive);
-	do_running_agg(p, psd, iOUT(6, pd), Globals.currIter, vo->veg[SW_SHRUB].biolive);
-	do_running_agg(p, psd, iOUT(7, pd), Globals.currIter, vo->veg[SW_TREES].biolive);
-	do_running_agg(p, psd, iOUT(8, pd), Globals.currIter, vo->veg[SW_FORBS].biolive);
-	do_running_agg(p, psd, iOUT(9, pd), Globals.currIter, biolive_total);
-
-	do_running_agg(p, psd, iOUT(10, pd), Globals.currIter,
-		v->veg[SW_GRASS].co2_multipliers[BIO_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(11, pd), Globals.currIter,
-		v->veg[SW_SHRUB].co2_multipliers[BIO_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(12, pd), Globals.currIter,
-		v->veg[SW_TREES].co2_multipliers[BIO_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(13, pd), Globals.currIter,
-		v->veg[SW_FORBS].co2_multipliers[BIO_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(14, pd), Globals.currIter,
-		v->veg[SW_GRASS].co2_multipliers[WUE_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(15, pd), Globals.currIter,
-		v->veg[SW_SHRUB].co2_multipliers[WUE_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(16, pd), Globals.currIter,
-		v->veg[SW_TREES].co2_multipliers[WUE_INDEX][SW_Model.simyear]);
-	do_running_agg(p, psd, iOUT(17, pd), Globals.currIter,
-		v->veg[SW_FORBS].co2_multipliers[WUE_INDEX][SW_Model.simyear]);
+		biomass_total += vo->veg[k].biomass * v->veg[k].cov.fCover;
+		litter_total += vo->veg[k].litter * v->veg[k].cov.fCover;
+		biolive_total += vo->veg[k].biolive * v->veg[k].cov.fCover;
+	}
+
+	// fCover for NVEGTYPES plus bare-ground
+	do_running_agg(p, psd, iOUT(0, pd), Globals.currIter, v->bare_cov.fCover);
+	i = 1;
+	ForEachVegType(k)
+	{
+		do_running_agg(p, psd, iOUT(i + k, pd), Globals.currIter,
+			v->veg[k].cov.fCover);
+	}
+
+	// biomass (g/m2 as component of total) for NVEGTYPES plus totals and litter
+	do_running_agg(p, psd, iOUT(i + NVEGTYPES, pd), Globals.currIter,
+		biomass_total);
+	i += NVEGTYPES + 1;
+	ForEachVegType(k) {
+		do_running_agg(p, psd, iOUT(i + k, pd), Globals.currIter,
+			vo->veg[k].biomass * v->veg[k].cov.fCover);
+	}
+	do_running_agg(p, psd, iOUT(i + NVEGTYPES, pd), Globals.currIter,
+		litter_total);
+

```

Comment:
why did you choose not to export litter for each veg type here?

Reply:
@kpalmqui I thought that adding four additional columns of output does not add new information and only results in the output files being even larger. 

This is because litter does not change with simulation time in `SOILWAT2` standalone runs, i.e., inputs = outputs --> no use for such runs. 

In `STEPWAT2` runs, litter changes from year-to-year, but `STEPWAT2` currently does not vary amounts of litter between functional types, i.e., total litter output is sufficient for `STEPWAT2` runs. See `STEPWAT2` litter inputs (https://github.com/DrylandEcology/STEPWAT2/blob/master/testing.sagebrush.master/Stepwat_Inputs/Input/sxw/sxwprod_v2.in) and see corresponding variable `_prod_litter` which does not vary by vegetation type or by resource group). However, I would add litter output for each vegetation type if you still want to?

Reply:
@dschalep no your decision makes sense given the above explanation. thanks!